zerver/tests/test_event_queue.py

```diff
@@ -608,6 +608,33 @@ def test_stream_push_notify_stream_specific_setting_with_muted_topic(self) -> No
                 already_notified={"email_notified": False, "push_notified": False},
             )
 
+    def test_stream_push_notify_stream_specific_setting_with_unmuted_topic_and_muted_stream(
+        self,
+    ) -> None:
+        # Push notification should be sent
+        self.change_subscription_properties({"push_notifications": True, "is_muted": True})
```

Comment:
Might make sense to do `email_notifications` true as well?

Reply:
I think it is not necessary to set either `stream.push_notifs` or `stream.email_notifs`. We already test elsewhere than notifications are not sent when `stream.is_muted = True` (even if `stream.push_notifications = True | stream.email_notifications = True`).
So here, just setting `stream.is_muted = True` and then checking that notifications are still sent if the topic is unmuted should be sufficient, I think.

Reply:
@abhijeetbodas2001 right!

I have updated the test.

Reply:
**4 new test cases are added.**

[Details (comment)](https://github.com/zulip/zulip/pull/24236#discussion_r1097780584)